RNA-seq/00c-tximeta_other_species.Rmd

```diff
@@ -0,0 +1,213 @@
+    "danio_rerio/cdna/Danio_rerio.GRCz11.cdna.all.fa.gz" 
+    ),
+  paste0(
+    fasta_base_url, 
+    # Replace the non-coding RNA fasta file path so that non-coding genes are also annotated
+    "danio_rerio/ncrna/Danio_rerio.GRCz11.ncrna.fa.gz" 
+    )
+)
+```
+
+Now we can follow a similar series of steps to find the file path for our species for the Gene Transfer File (GTF) using this address: 
+http://ftp.ensembl.org/pub/release-98/gtf/
+
+```{r}
+gtf_base_url <- "ftp://ftp.ensembl.org/pub/release-98/gtf/"
+
+gtf_files <- paste0(
+  gtf_base_url, 
+  # Replace the file path below to the species-specific gtf file
+  "danio_rerio/Danio_rerio.GRCz11.103.gtf.gz"
+  )
+```
+
+Create a `linkedTxome` with the links to the fasta files, gtf file, and salmon index specific to our species. 
+
+```{r}
+makeLinkedTxome(indexDir = index_dir,
+                source = "Ensembl",
+                organism = "Danio rerio", # Replace this with the species name
+                release = "98", # Replace this with the release number you've obtained the files from
+                genome = "GRCz11.103", # Replace with the genome version noted in the gtf file
+                fasta = fasta_files,
+                gtf = gtf_files)
+```
+
+Now we can continue on with the setup and `tximeta` steps we've used for human and mouse. 
+
+## File names
+
+```{r sf_files}
+# Let's look at the full path for the quant.sf files
+sf_files
+```
+
+Let's extract the _sample_ names from the **file paths** using the `stringr` package.
+
+```{r sample_names}
+sample_names <- stringr::word(sf_files, -2, sep = "/")
+sample_names
+```
+
+## Set up metadata
+
+```{r names_sf_files}
+coldata <- data.frame(files = sf_files, 
+                      names = sample_names)
+```
+
+We have more information about these samples stored in the metadata file that we will also want stored in `coldata`. 
+Let's read in the sample metadata from the TSV file and do a bit of set up. 
+We really only want two columns from this metadata file, so we will use a `dplyr::select()` to keep only those. 
+
+```{r sample_meta_df}
+# Read in the sample metadata TSV file and have a look
+sample_meta_df <- readr::read_tsv(meta_file) %>% 
+  # Select the two columns we need here
+  dplyr::select(refinebio_accession_code, refinebio_title)
+
+# Take a look at the preview
+sample_meta_df
+```
+
+We'll want this information to be added to the `coldata`, which we can do by using a join function to match up the rows between the two data frames and combine them. 
+
+```{r join-sample_meta_df}
+coldata <- coldata %>%
+  dplyr::inner_join(sample_meta_df, by = c("names" = "refinebio_accession_code"))
+
+coldata
+```
+
+## Import expression data with `tximeta`
+
+Using the `coldata` data frame and the `linkedTxome` we set up, now run the `tximeta()` per usual.
+
+```{r tximeta}
+txi_data <- tximeta(coldata)
+```
+
+If you haven't created this linkedTxome before, in the console window, you'll be asked: 
```

Comment:
Is there any way to trigger this step before calling `tximeta()`? I would be confused having thought that I had already created the `linkedTxome` when I called `makeLinkedTxome()`.

If the answer is to the above question is `no` I would suggest the following edit:
```suggestion
If you haven't used this linkedTxome before, in the console window, you'll be asked: 
```